lib/puppet/functions/vault_lookup/lookup.rb

```diff
@@ -1,10 +1,15 @@
 Puppet::Functions.create_function(:'vault_lookup::lookup') do
   dispatch :lookup do
     param 'String', :path
-    param 'String', :vault_url
+    optional_param 'String', :vault_url
   end
 
-  def lookup(path, vault_url)
+  def lookup(path, vault_url = nil)
+    if vault_url.nil?
+      debug 'No Vault address was set on function, defaulting to value from VAULT_ADDR env value'
+      vault_url = ENV['VAULT_ADDR']
```

Comment:
What if the environment variable doesn’t exist? Fail with some helpful error message here instead of exploding in line 13?

Reply:
Fixed

Reply:
And added new spec for this paricular error 😄